StyleCop.Analyzers/StyleCop.Analyzers.Test.CSharp9/DocumentationRules/SA1649CSharp9UnitTests.cs

```diff
@@ -3,9 +3,99 @@
 
 namespace StyleCop.Analyzers.Test.CSharp9.DocumentationRules
 {
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.CodeAnalysis;
     using StyleCop.Analyzers.Test.CSharp8.DocumentationRules;
+    using StyleCop.Analyzers.Test.Verifiers;
+    using Xunit;
+    using static StyleCop.Analyzers.Test.Verifiers.CustomDiagnosticVerifier<StyleCop.Analyzers.DocumentationRules.SA1649FileNameMustMatchTypeName>;
 
     public partial class SA1649CSharp9UnitTests : SA1649CSharp8UnitTests
     {
+        [Fact]
+        [WorkItem(3967, "https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/3967")]
+        public async Task VerifyTopLevelStatementsOnlyDoNotReportAsync()
+        {
+            var testCode = @"System.Console.WriteLine();
+";
+
+            var test = new StyleCopCodeFixVerifier<StyleCop.Analyzers.DocumentationRules.SA1649FileNameMustMatchTypeName, StyleCop.Analyzers.DocumentationRules.SA1649CodeFixProvider>.CSharpTest()
+            {
+                TestState =
+                {
+                    OutputKind = OutputKind.ConsoleApplication,
+                    Sources = { ("Program.cs", testCode) },
```

Comment:
I assume the name Program.cs is not mandated. Make this a Theory with different file names, to have tests in place to check that?